test/csuite/wt4105_large_doc_small_upd/main.c

```diff
@@ -32,7 +32,7 @@
 static const char *const uri = "table:large";
 
 #define DATASIZE (1024 * 1024)
-#define MODIFY_COUNT (1024)
+#define MODIFY_COUNT (1000)
```

Comment:
Is this a meaningful change?

Reply:
Yes. This test performs small modifications on a large update and none of the update is globally visible. Once the number of updates is greater than 1000, we trigger the forced eviction of the page, and the next cursor operation that is performing the forced eviction takes more time than expected, and the test fails.

To avoid triggering the forced eviction, I reduced the number of updates and the test passed continuously.

Reply:
Can you explain why this change makes that experience worse? I'm surprised it is related.

Reply:
The existing obsolete check gets avoided when the transaction state is not moved forward. That was preventing this page from being marked for urgent eviction. But when I removed those checks, it led to an unnecessary forced eviction, and the test failed.

As I added back the transaction state logic, I reverted these changes.